libnavui-speedlimit/src/main/java/com/mapbox/navigation/ui/speedlimit/model/PostedAndCurrentSpeedFormatter.kt

```diff
@@ -0,0 +1,61 @@
+package com.mapbox.navigation.ui.speedlimit.model
+
+import com.mapbox.navigation.base.formatter.UnitType
+import com.mapbox.navigation.base.speed.model.SpeedUnit
+import com.mapbox.navigation.ui.base.formatter.ValueFormatter
+import kotlin.math.roundToInt
+
+/**
+ * Formats posted speed limit and current speed data to [Int] for displaying to a user.
+ */
+class PostedAndCurrentSpeedFormatter : ValueFormatter<SpeedData, Int> {
+
+    /**
+     * Formats posted speed limit and current speed data.
+     *
+     * @param update a state containing [SpeedData]
+     *
+     * @return a formatted string
+     */
+    override fun format(update: SpeedData): Int {
+        return getPostedSpeedLimit(update.speed, update.fromUnit, update.toUnit)
+    }
+
+    private fun getPostedSpeedLimit(
+        speed: Double,
+        fromUnit: SpeedUnit,
+        toUnit: UnitType
+    ): Int {
+        return when (fromUnit) {
+            SpeedUnit.METERS_PER_SECOND -> {
+                when (toUnit) {
+                    UnitType.IMPERIAL -> speed.metersPerSecondToMilesPerHour()
+                    UnitType.METRIC -> speed.metersPerSecondToKilometersPerHour()
+                }.roundToInt()
+            }
+            SpeedUnit.KILOMETRES_PER_HOUR -> {
+                when (toUnit) {
+                    UnitType.IMPERIAL -> 5 * (speed.kilometersToMiles() / 5).roundToInt()
```

Comment:
What's this magic with the fives?

Reply:
That rounds a speed value to the nearest number that is divisible by 5.

Reply:
Why should it be divisible by 5?